### DISCUSSION TITLE
Help with designing schema of a document represented in tinybase

### DISCUSSION THREAD

**Discussion by vineetdigit** [tinyplex/tinybase#65] [General]
I'm little confused how to design schema for a document with tinybase. For the sake of simplicity, let's assume that the document is as follows

```
{
  title: String,
  pages: [Number|JSON|String]
}
```

From what I understand, tinybase allows either key=value pairs or tables. So in order to represent the document, do I need to store title as a key=value pair and pages as a table? If yes, then how do I create one-to-many relationship from the document to the pages?

Any help?

thx

**Comment by jamesgpearce** [comment]
I would try to keep everything in tables - since these feels like a relational model, and you can maybe use the relationships module to connect the documents to the pages.

So maybe something like this?

```js
const store = createStore();
store.setTables({
  documents: {
    docA: {title: 'The title for doc A'},
    docB: {title: 'The title for doc B'},
    // ...
  },
  pages: {
    pageA1: {doc: 'docA', order: 1, content: 'blah blah'},
    pageA2: {doc: 'docA', order: 2, content: 'blah blah'},
    pageB1: {doc: 'docB', order: 1, content: 'blah blah'},
    pageB2: {doc: 'docB', order: 2, content: 'blah blah'},
    // ...
  },
});
```

The relationships module could then be used something like:

```js
const relationships = createRelationships(store);
relationships.setRelationshipDefinition(
  'pagesToDocuments', // relationshipId
  'pages',            // localTableId to link from
  'documents',        // remoteTableId to link to
  'doc',              // cellId containing remote key
);

console.log(relationships.getRemoteRowId('pagesToDocuments', 'pageA2'));
// -> 'docA'
console.log(relationships.getLocalRowIds('pagesToDocuments', 'docB'));
// -> ['pageB1', 'pageB2']
```
